pinconfigs/README.md

```diff
@@ -0,0 +1,154 @@
+# GD32 pin configurations
+
+This directory contains a set of files describing valid pin configurations for
+multiple GD32 devices, for example, that `USART0_TX` signal can only be mapped
+to `PA0` and `PC8`. These configurations can be used to generate valid pin
+mappings.
+
+GD32 devices use two distinct models to configure signal multiplexing. In
+some devices, like GD32F103XX, each pin has one or more associated alternate
+function. These alternate functions can be sometimes changed using the remap
+functionality, which allows routing all signals from a peripheral to a
+different set of pins. For this reason we will name the model used by such
+devices *AFIO model*. Some other devices, like GD32F450XX, have a simpler
+and more advanced model where each pin has a fixed list of alternate functions.
+Similarly, we will name the model used by such devices *AF model*. The datasheet
+(not reference manual) can be used to determine which model is used by a device.
+If the pin definition table mentions *remaps*, it means the device uses the
+*AFIO model*. Alternate function tables (AF0..15) will only be available on
+devices using the *AF model*.
+
+In the following sections the expected file format is described.
+
+## Common configuration
+
+Some fields are common independently of the pin configuration model.
+The commented example below describes common fields in detail.
+
+- `model` (required): Choose between `afio` or `af`
+- `series` (required): Series name, e.g. gd32vf103
+- `variants` (required): Each variant has a different set of valid pin
```

Comment:
For GD32F405/407, BGA100 and LQPF100 both have the 'V' pincode, but BGA100 doesn't have the PD8 pin. It can be check at datasheet 2.3, 2.6 pins related section. GD32F403 doesn't have this issue.

I think it's a GigaDevice mistake, they told GD32F405/407 BGA100 have 82 IO too, but actually is 81.

Reply:
We can probably document that in the config file with a comment (or allow inserting comments from the config file to the generated headers).